tests/vrf03.rs

```diff
@@ -278,4 +278,82 @@ mod test {
 
         Ok(())
     }
+
+    use proptest::prelude::*;
+    use vrf_dalek::constants::SEED_SIZE;
+    use vrf_dalek::errors::VrfError;
+
+    fn secret_public_keys() -> impl Strategy<Value = ([u8; SEED_SIZE], PublicKey03)> {
+        proptest::string::bytes_regex("[[:ascii:]]{32}")
+            .unwrap()
+            .prop_map(|vec| {
+                let mut bytes = [0u8; 32];
+                bytes.copy_from_slice(&vec);
+                let sk = SecretKey03::from_bytes(&bytes);
+                (bytes, PublicKey03::from(&sk))
+            })
+    }
+
+    fn payload() -> impl Strategy<Value = Vec<u8>> {
+        proptest::string::bytes_regex("[[:ascii:]]{0,254}").unwrap()
+    }
+
+    proptest! {
+        #[test]
+        fn public_key_has_32_bytes((_,pk) in secret_public_keys()) {
+            let pk_bytes = PublicKey03::to_bytes(pk);
+            prop_assert!(pk_bytes.len() == 32);
+        }
+
+        #[test]
+        fn proof_has_80_bytes(((sk_bytes,pk),alpha) in (secret_public_keys(), payload())) {
+            let sk = SecretKey03::from_bytes(&sk_bytes);
+            let proof = VrfProof03::generate(&pk, &sk, &alpha);
+            let proof_bytes = VrfProof03::to_bytes(&proof);
+            prop_assert!(proof_bytes.len() == 80);
+        }
+
+        #[test]
+        fn proof_to_hash_equals_verify_result(((sk_bytes,pk),alpha) in (secret_public_keys(), payload())) {
+            let sk = SecretKey03::from_bytes(&sk_bytes);
+            let proof = VrfProof03::generate(&pk, &sk, &alpha);
+            let hash_from_proof = VrfProof03::proof_to_hash(&proof);
+            let verify_res = proof.verify(&pk, &alpha);
+            prop_assert!(verify_res.is_ok());
+            prop_assert!(verify_res.unwrap() == hash_from_proof);
+        }
+
+        #[test]
+        fn proofs_from_different_alpha_never_equals(((sk_bytes,pk),alpha1,alpha2) in (secret_public_keys(), payload(), payload())) {
+            let sk = SecretKey03::from_bytes(&sk_bytes);
+            let proof1 = VrfProof03::generate(&pk, &sk, &alpha1);
+            let proof2 = VrfProof03::generate(&pk, &sk, &alpha2);
+            prop_assert!(VrfProof03::to_bytes(&proof1) != VrfProof03::to_bytes(&proof2));
+        }
+
+        #[test]
+        fn proofs_from_different_secret_keys_never_equals(((sk_bytes1,pk1),(sk_bytes2,pk2),alpha) in (secret_public_keys(), secret_public_keys(), payload())) {
```

Comment:
Although it's theoretical, I would keep the payload constant here because it could be the case a specific combination of different keys and payload yield same proof.

Reply:
:+1: